backend/src/services/service-token/service-token-service.ts

```diff
@@ -154,21 +154,21 @@ export const serviceTokenServiceFactory = ({
   };
 
   const fnValidateServiceToken = async (token: string) => {
-    const [, TOKEN_IDENTIFIER, TOKEN_SECRET] = <[string, string, string]>token.split(".", 3);
-    const serviceToken = await serviceTokenDAL.findById(TOKEN_IDENTIFIER);
+    const [, tokenIdentifier, tokenSecret] = <[string, string, string]>token.split(".", 3);
+    const serviceToken = await serviceTokenDAL.findById(tokenIdentifier);
 
-    if (!serviceToken) throw new UnauthorizedError();
+    if (!serviceToken) throw new NotFoundError({ message: "Service token not found" });
     const project = await projectDAL.findById(serviceToken.projectId);
 
-    if (!project) throw new UnauthorizedError({ message: "Service token project not found" });
+    if (!project) throw new NotFoundError({ message: "Service token project not found" });
 
     if (serviceToken.expiresAt && new Date(serviceToken.expiresAt) < new Date()) {
       await serviceTokenDAL.deleteById(serviceToken.id);
-      throw new UnauthorizedError({ message: "failed to authenticate expired service token" });
+      throw new ForbiddenRequestError({ message: "Service token has expired" });
```

Comment:
The user is not authenticated at all here so we should be throwing UnauthorizedError. The credentials they provided are expired